From 8348d44b4b8a4d05d5ab16c75cbfbdc3664e1242 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 30 Jun 2016 13:38:39 +0800
Subject: [PATCH 1105/5242] MLK-12948 ARM: dts: imx7d-sdb: change the hardware
 reset gpio for mipi dsi

commit  e7e5e62db0ca7311a124f5157d7c772b5d50eca5 from
https://source.codeaurora.org/external/imx/linux-imx.git

Change the hardware reset gpio to 'GPIO6_IO15' for mipi dsi to
allow fec2 and mipi dsi can run at the same time. This needs
some hardware rework as follows:
"
    1. Replace R631 with 100K resistor;
    2. Remove D14, D24;
    3. Solder the Cathode of the diode to R471,
       you can use BAT54HT1(ONSEMI) or NSR0320MW2T1G(ONSEMI);
    4. Solder the wire to the Anode end of the diode;
    5. Scrape the solder mask(Green oil)  of the MIPI Reset via,
       then solder the end of the wire to the via.
"

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-sdb-mipi-dsi.dts |    8 ++------
 arch/arm/boot/dts/imx7d-sdb.dts          |    6 ++++++
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-sdb-mipi-dsi.dts b/arch/arm/boot/dts/imx7d-sdb-mipi-dsi.dts
index 71d6684..bbfca36 100644
--- a/arch/arm/boot/dts/imx7d-sdb-mipi-dsi.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-mipi-dsi.dts
@@ -11,23 +11,19 @@
 / {
 	mipi_dsi_reset: mipi-dsi-reset {
 		compatible = "gpio-reset";
-		reset-gpios = <&gpio1 4 GPIO_ACTIVE_LOW>;
+		reset-gpios = <&gpio6 15 GPIO_ACTIVE_LOW>;
 		reset-delay-us = <1000>;
 		#reset-cells = <0>;
 	};
 };
 
-&fec2 {
-	status = "disabled";
-};
-
 &lcdif {
 	disp-dev = "mipi_dsi_samsung";
 };
 
 &mipi_dsi {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_enet2_epdc0_en>;
+	pinctrl-0 = <&pinctrl_mipi_dsi_reset>;
 	lcd_panel = "TRULY-WVGA-TFT3P5581E";
 	resets = <&mipi_dsi_reset>;
 	status = "okay";
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index a88d32f..9b3f22e 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -783,6 +783,12 @@
 			>;
 		};
 
+		pinctrl_mipi_dsi_reset: mipi_dsi_reset_grp {
+			fsl,pins = <
+				MX7D_PAD_SAI1_TX_DATA__GPIO6_IO15 0x1b
+			>;
+		};
+
 		pinctrl_ecspi3: ecspi3grp {
 			fsl,pins = <
 				MX7D_PAD_SAI2_TX_SYNC__ECSPI3_MISO	0x2
-- 
1.7.9.5

